### PR TITLE
Support trim operation on a volume with multiple mountpoints

### PR DESCRIPTION
**Pull request by derekbit**
Longhorn/longhorn#836

Signed-off-by: Derek Su <derek.su@suse.com>

### CODE REVIEW

**Review by innobead — APPROVED** [review]
LGTM

**Comment by innobead** [comment]
@Mergifyio backport v1.4.x

File: util/util.go [hunk 1]
```diff
@@ -912,17 +912,20 @@ func TrimFilesystem(volumeName string) error {
...
 		return fmt.Errorf("cannot find volume %v mount info on host: %v", volumeName, err)
 	}
 
 	mountList := strings.Split(strings.TrimSpace(mountOutput), "\n")
-	if len(mountList) != 1 {
-		return fmt.Errorf("zero or multiple mount points for volume %s: %+v", volumeName, mountList)
+
+	var mountpoint string
+	for _, m := range mountList {
+		mountpoint = m
```
Comment (shuo-wu) [inline util/util.go:924]: Should we check if this mount point is the valid one?

Comment (derekbit) [reply util/util.go:924]: Would be better. But how to check it is a valid one?

Comment (shuo-wu) [reply util/util.go:924]: Function `ensureMountPoint` in the pkg `csi` already implements that

Comment (innobead) [reply util/util.go:924]: `ensureMountPoint` will try to clean up the corrupted mount point so it's not just a validation behavior, so probably we need to have a separate one for `mount.IsCorruptedMnt(err)` only.

https://github.com/longhorn/longhorn-manager/blob/db24c350211c524eab04afc702fdc0497aabeed6/csi/util.go#L216-L220

Comment (derekbit) [reply util/util.go:924]: I'm thinking if we need to check if the mountpoint is corrupted. If it is corrupted, the fstrim should return error.

Comment (shuo-wu) [reply util/util.go:924]: Maybe we can have a try.

Comment (shuo-wu) [reply util/util.go:924]: OK. It shows error `fstrim: vol: FITRIM ioctl failed: Input/output error`.

Here it's better to figure out which mount point is the valid one then apply `fstrim`

Comment (derekbit) [reply util/util.go:924]: Then, let me add a check.

**Pull request merged**